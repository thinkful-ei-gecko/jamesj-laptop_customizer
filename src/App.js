import React, { Component } from 'react'

// Normalizes string as a slug - a string that is safe to use
// in both URLs and html attributes

import './App.css'
import CustomizeList from './CustomizeList'
import CartList from './CartList'

// This object will allow us to
// easily convert numbers into US dollar values
const USCurrencyFormat = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
})

class App extends Component {
  constructor() {
    super()
    this.state = {
      features: {
        Processor: [
          {
            name: '17th Generation Intel Core HB (7 Core with donut spare)',
            cost: 700,
          },
          {
            name: 'Professor X AMD Fire Breather with sidewinder technology',
            cost: 1200,
          },
        ],
        'Operating System': [
          {
            name: 'Ubuntu Linux 16.04',
            cost: 200,
          },
          {
            name: 'Bodhi Linux',
            cost: 300,
          },
        ],
        'Video Card': [
          {
            name: 'Toyota Corolla 1.5v',
            cost: 1150.98,
          },
          {
            name: 'Mind mild breeze 2000',
            cost: 1345,
          },
        ],
        Display: [
          {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500,
          },
          {
            name: '15.3" HGTV (3840 x 2160) Home makeover edition',
            cost: 1400,
          },
        ],
      },
      selected: {
        Processor: {
          name: '17th Generation Intel Core HB (7 Core with donut spare)',
          cost: 700,
        },
        'Operating System': {
          name: 'Ubuntu Linux 16.04',
          cost: 200,
        },
        'Video Card': {
          name: 'Toyota Corolla 1.5v',
          cost: 1150.98,
        },
        Display: {
          name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
          cost: 1500,
        },
      },
    }
  }

  updateFeature = (feature, newValue) => {
    const selected = Object.assign({}, this.state.selected)
    selected[feature] = newValue
    this.setState({
      selected,
    })
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1>ELF Computing | Laptops</h1>
        </header>
        <main>
          <form className="main__form">
            <h2>Customize your laptop</h2>
            <CustomizeList
              features={this.state.features}
              formatUS={USCurrencyFormat}
              selected={this.state.selected}
              updateFeature={(feature, newValue) =>
                this.updateFeature(feature, newValue)
              }
            />
          </form>
          <section className="main__summary">
            <h2>Your cart</h2>
            <CartList
              formatUS={USCurrencyFormat}
              selected={this.state.selected}
            />
          </section>
        </main>
      </div>
    )
  }
}

export default App
