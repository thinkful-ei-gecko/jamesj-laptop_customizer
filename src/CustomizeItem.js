import React from 'react';
import slugify from 'slugify';

export default function CustomizeItem(props) { 
  console.log('CustomizeItem props are:',props)
  return (
  <div key={props.itemHash} className="feature__item">
    <input
      type="radio"
      id={props.itemHash}
      className="feature__option"
      name={slugify(props.feature)}
      checked={props.name === props.selected[props.feature].name}
      onChange={e => props.updateFeature(props.feature, props.item)}
    />
    <label htmlFor={props.itemHash} className="feature__label">
      {props.name} ({props.USCurrencyFormat.format(props.cost)})
    </label>
  </div>
)}