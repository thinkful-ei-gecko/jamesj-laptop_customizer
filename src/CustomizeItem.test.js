import React from 'react'
import ReactDOM from 'react-dom'
import CustomizeItem from './CustomizeItem'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { USCurrencyFormat, updateFeature } from './App'

describe('CustomizeItem component', () => {
  const item = {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700,
    }
  const selected = {
    Processor: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700,
    },
  }
  it('should render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <CustomizeItem
        key="{itemHash}"
        itemHash="{itemHash}"
        feature="Processor"
        formatUS={USCurrencyFormat}
        item={item}
        {...item}
        selected={selected}
        updateFeature={updateFeature}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders same UI', () => {
    const wrapper = shallow(
      <CustomizeItem
        key="{itemHash}"
        itemHash="{itemHash}"
        feature="Processor"
        formatUS={USCurrencyFormat}
        item={item}
        {...item}
        selected={selected}
        updateFeature={updateFeature}
      />
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
