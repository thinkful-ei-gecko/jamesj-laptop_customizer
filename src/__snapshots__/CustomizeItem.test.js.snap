// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomizeItem component renders same UI 1`] = `
<div
  className="feature__item"
  key="{itemHash}"
>
  <input
    checked={true}
    className="feature__option"
    id="{itemHash}"
    name="Processor"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor="{itemHash}"
  >
    17th Generation Intel Core HB (7 Core with donut spare)
     (
    $700.00
    )
  </label>
</div>
`;
